src/GeneratedHydrator/CodeGenerator/Visitor/HydratorMethodsVisitor.php

```diff
@@ -164,7 +164,7 @@ private function replaceHydrate(ClassMethod $method)
 
         $bodyParts = array();
         foreach ($this->visiblePropertyMap as $propertyName) {
-            $bodyParts[] = "if (isset(\$data['" . $propertyName . "'])) {";
+            $bodyParts[] = "if (isset(\$data['$propertyName']) || \$object->$propertyName !== null && array_key_exists('$propertyName', \$data)) {";
```

Comment:
Please don't use string interpolation - made me super-confused while reading this diff